wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -557,3 +559,78 @@ function inject_other_filters( $key ) {
 		printf( '<input type="hidden" name="s" value="%s" />', esc_attr( $wp_query->query['s'] ) );
 	}
 }
+
+/**
+ * Filter the block data for the student course list.
+ *
+ * @param array $parsed_block The parsed block data.
+ * @return array The updated block data.
+ */
+function filter_student_course_list( $parsed_block ) {
+	if (
+		'core/query' !== $parsed_block['blockName'] ||
+		'course' !== ( $parsed_block['attrs']['query']['postType'] ?? '' ) ||
+		! isset( $parsed_block['attrs']['queryId'] ) ||
+		! is_page( 'my-courses' )
+	) {
+		return $parsed_block;
+	}
+
+	if ( ! array_key_exists( 'exclude', $parsed_block['attrs']['query'] ) || ! is_array( $parsed_block['attrs']['query']['exclude'] ) ) {
+		$parsed_block['attrs']['query']['exclude'] = array();
+	}
+
+	$exclude_ids = get_course_ids_to_be_excluded( $parsed_block['attrs']['queryId'] );
+
+	$parsed_block['attrs']['query']['exclude'] = array_merge(
+		$parsed_block['attrs']['query']['exclude'],
+		$exclude_ids
+	);
+
+	return $parsed_block;
+}
+
+/**
+ * Get the course IDs to be excluded from the query.
+ */
+function get_course_ids_to_be_excluded( $query_id ): array {
```

Comment:
$query_id isn't used
```suggestion
function get_course_ids_to_be_excluded(): array {
```

Reply:
bec4e5c